### ISSUE TITLE
[c_api] More friendly error messages 

### ISSUE THREAD

**Issue by YuCrazing** [taichi-dev/taichi#6406]
**Describe the bug**
When copying host data into a ndarray without host accessibility,  there is an obscure error message: `Memory is not mapped`. I think it is better to let this error message more clearer. For example: `This ndarray is not accessible from the host. Please assure host_access=true.`

**To Reproduce**

```c++
ti::NdArray<int> arr = runtime_.allocate_ndarray<int>(std::vector<u_int32_t>{5}, {}, /*host_access=false*/);

std::vector<int> arr_host = {1.0, 1.0, 1.0, 1.0, 1.0};
arr.write(arr_host);
```

error message:
```c++
[E 10/18/22 15:21:30.564 700888] [vulkan_device.cpp:unmap@1590] Memory is not mapped


[W 10/18/22 15:21:30.564 700888] [taichi_core_impl.cpp:ti_set_last_error@130] C-API error: (invalid state) [vulkan_device.cpp:unmap@1590] Memory is not mapped
```

**Comment by ailzhang** [comment]
@PENGUINLIONG IIRC we planned to change read/write to use staging buffer instead of map/unmap, is that correct understanding?

**Comment by PENGUINLIONG** [comment]
> @PENGUINLIONG IIRC we planned to change read/write to use staging buffer instead of map/unmap, is that correct understanding? 

y, the error message can be changed for now tho

**Comment by PENGUINLIONG** [comment]
Solved by #6417
